logger/logger.go

```diff
@@ -77,19 +122,27 @@ func newLogWriter(config LogConfig) io.Writer {
 	if config.Quiet {
 		wlog.SetLevel(wlog.ERROR)
 	}
+	var logWriter io.Writer
+	if logCreator, ok := loggerRegistry[config.LogTarget]; ok {
+		logWriter, _ = logCreator.createLogger(config)
+	}
+	if logWriter == nil {
+		logWriter, _ = (&telegrafLogCreator{}).createLogger(config)
+	}
 
-	var writer io.Writer
-	if config.Logfile != "" {
-		var err error
-		if writer, err = rotate.NewFileWriter(config.Logfile, config.RotationInterval.Duration, config.RotationMaxSize.Size, config.RotationMaxArchives); err != nil {
-			log.Printf("E! Unable to open %s (%s), using stderr", config.Logfile, err)
-			writer = os.Stderr
-		}
-	} else {
-		writer = os.Stderr
+	if closer, isCloser := actualLogger.(io.Closer); isCloser {
+		closer.Close()
```

Comment:
Not sure if its an issue, but can you double check that we don't close stderr?  Also, we may not want to close the logfile as it could mean we move back to stderr logging on reload.

Reply:
Good point. But actual check was needed in the `telegrafLog.Close`, which is responsible for closing internal log.